infra/cron/src/jobs/consistency.js

```diff
@@ -0,0 +1,100 @@
+const checker = require('@origin/consistency')
+const fs = require('fs')
+const tmp = require('tmp')
+const sendgridMail = require('@sendgrid/mail')
+const esmImport = require('esm')(module)
+const contractsContext = esmImport('@origin/graphql/src/contracts').default
+const { setNetwork } = esmImport('@origin/graphql/src/contracts')
+const logger = require('../logger')
+
+function todaysDateString() {
+  const now = new Date()
+  const year = now.getUTCFullYear()
+  const month = String(now.getUTCMonth()).padStart(2, '0')
+  const day = String(now.getUTCDate()).padStart(2, '0')
+  return `${year}-${month}-${day}`
+}
+
+/**
+ * A job to run consistency checker and send an E-mail with results
+ * @param job
+ * @returns {Promise<*>}
+ */
+async function ConsistencyCheckJob(job) {
+  logger.info(`Processing ConsistencyCheckJob id=${job.id} data= ${job.data}`)
+  job.progress(0)
+
+  if (typeof process.env.PROVIDER_URL === 'undefined') {
+    throw new Error('Env var PROVIDER_URL must be set')
+  }
+  if (typeof process.env.EMAILS_TO === 'undefined') {
+    throw new Error('Warning: EMAILS_TO env var is not set')
+  }
+  if (typeof process.env.SENDGRID_API_KEY === 'undefined') {
+    throw new Error('Env var SENDGRID_API_KEY must be set')
+  }
+  if (typeof process.env.SENDGRID_FROM_EMAIL === 'undefined') {
+    logger.warn('Warning: SENDGRID_FROM_EMAIL env var is not set')
+  }
+
+  const logFile = tmp.fileSync()
+  const network = process.env.NETWORK || 'mainnet'
+
+  setNetwork(network)
+  if (!contractsContext.web3) {
+    throw new Error('web3 not initialized')
+  }
+  console.log(`Logging to file ${logFile.name}`)
```

Comment:
nit: use logger rather than console

Reply:
Weird, I remember explicitly doing that.  Thanks